lighthouse-core/audits/autocomplete.js

```diff
@@ -43,22 +194,80 @@ class AutocompleteAudit extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts.FormInput} input
+   * @return {{hasValidTokens: Boolean, inValidOrder?: Boolean}}
+   */
+  static checkAttributeValidity(input) {
+    if (!input.autocomplete.attribute) return {hasValidTokens: false};
+    const tokenArray = input.autocomplete.attribute.split(' ');
+    for (const token of tokenArray) {
+      // A `section-` prefix indicates a unique autofill scope.
+      // https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#:~:text=section-
+      if (token.slice(0, 8) === 'section-') continue;
+      if (validAutocompleteTokens.includes(token)) continue;
+      return {hasValidTokens: false};
+    }
+    // If all autocomplete tokens are valid but there is still no property attribute, then that means the tokens are out of order.
+    // https://cloudfour.com/thinks/autofill-what-web-devs-should-know-but-dont/#all-the-tokens
+    if (!input.autocomplete.property) return {hasValidTokens: true, inValidOrder: true};
+    return {hasValidTokens: true};
+  }
+
   /**
    * @param {LH.Artifacts} artifacts
    * @return {LH.Audit.Product}
    */
   static audit(artifacts) {
     const forms = artifacts.FormElements;
     const failingFormsData = [];
+    const warnings = [];
+    let notApplicable = 0;
+    let inputsCount = 0;
     for (const form of forms) {
       for (const input of form.inputs) {
-        if (!input.autocomplete) {
+        inputsCount += 1;
+        const autocomplete = this.checkAttributeValidity(input);
+        if (!autocomplete.hasValidTokens || autocomplete.inValidOrder) {
+          if (!input.autocomplete.prediction) {
+            notApplicable += 1;
+            continue;
+          }
+          if (noPrediction.includes(input.autocomplete.prediction) &&
+          !input.autocomplete.attribute) {
+            notApplicable += 1;
+            continue;
+          }
+          const snippetArray = input.snippet.split(' title=');
+          const snippet = snippetArray[0] + '>';
+          // @ts-ignore
+          let suggestion = autofillSuggestions[input.autocomplete.prediction];
+          // This is here to satisfy typescript because the possible null value of autocomplete.attribute is not compatible with Audit details.
+          if (!input.autocomplete.attribute) {
+            input.autocomplete.attribute = '';
+          }
+          if (input.autocomplete.attribute) {
+            warnings.push(str_(UIStrings.warningInvalid, {token: input.autocomplete.attribute,
+              snippet: snippet}));
+          }
+          if (autocomplete.inValidOrder) {
```

Comment:
I can't tell if `inValidOrder` means "this is in a valid order" or "this is in an invalid order". refactor to `isValidOrder` ?